File: dotnet/Clarian.Sdk/BalancesService.cs
namespace Clarian;

/// <summary>Workspace ledger balances.</summary>
public sealed class BalancesService
{
    private readonly ClarianClient _client;

    internal BalancesService(ClarianClient client) => _client = client;

    /// <summary>GET /account/balances.</summary>
    public async Task<IReadOnlyList<Balance>> ListAsync(CancellationToken cancellationToken = default)
    {
        var res = await _client.RequestAsync<BalancesResponse>(
            HttpMethod.Get,
            "/account/balances",
            null,
            null,
            cancellationToken).ConfigureAwait(false);
        return res.Balances ?? new List<Balance>();
    }
}

File: java/src/main/java/finance/clarian/sdk/BalancesService.java
package finance.clarian.sdk;

import java.util.Collections;
import java.util.List;
import java.util.Map;

/** Workspace ledger balances. */
public final class BalancesService {

    private final ClarianClient client;

    BalancesService(ClarianClient client) {
        this.client = client;
    }

    /** GET /account/balances. */
    @SuppressWarnings("unchecked")
    public List<Map<String, Object>> list() {
        Map<String, Object> res = client.request("GET", "/account/balances", null, null);
        Object bals = res.get("balances");
        if (bals instanceof List<?> list) {
            return (List<Map<String, Object>>) list;
        }
        return Collections.emptyList();
    }
}
